### PR TITLE
[WIP] Concatenate 3d with 4d images into a 4d image

### PR DESCRIPTION
**Pull request by dchorel**
I have not made any test because there's no 4D images in the atlas folder.
Tested by creating a concatenated image with 2 3D images from atlas folder and by concatenating this resulted image with an other 3D image.

### CODE REVIEW

**Review by mdesco — COMMENTED** [review]
Please test! This really does not work.

1) Name of the operation should be "concatenate" not "concat"

2) Does not work for me:
scil_image_math.py concatenate b0.nii.gz dwi.nii.gz test.nii.gz
https://www.dropbox.com/s/2org73bo311ujrf/testPR380.zip?dl=0

Can we create PR to concatenate gradient tables?

scil_concatenate_diffusion.py dwi1.nii.gz dwi1.bval dwi1.bvec  dwi2.nii.gz dwi2.bval dwi2.bvec

File: scilpy/image/operations.py [hunk 1]
```diff
@@ -83,6 +83,14 @@ def _validate_imgs(*imgs):
             raise ValueError('Not all inputs have the same shape!')
 
 
+def _validate_imgs_concat(*imgs):
+    """Make sure that all inputs are images."""
+    ref_img = imgs[-1]
```
Comment (frheault) [inline scilpy/image/operations.py:88]: ref_img isn't used?

File: scilpy/image/operations.py [hunk 2]
```diff
@@ -83,6 +83,14 @@ def _validate_imgs(*imgs):
...
+    for img in imgs:
+        if isinstance(img, nib.Nifti1Image):
```
Comment (frheault) [inline scilpy/image/operations.py:90]: if not isinstance(img, nib.Nifti1Image):

**Review by frheault — COMMENTED** [review]
Line 148: mask[data > 0] = 1
Must become:
if data.ndim == 4:
    mask[np.sum(data, axis=3).astype(bool) > 0] = 1
else:
    mask[data > 0] = 1
  
  It seems to work for me (I concatenated b0 + dwi + b0, b0 + b0 + dwi, dwi + dwi + b0, b0 + b0)

**Review by arnaudbore — APPROVED** [review]

**Pull request merged**